packages/monitor-v2/src/monitor-polymarket/MonitorProposalsOrderBook.ts

```diff
@@ -92,13 +92,25 @@ export async function monitorTransactionsProposedOrderBook(
     .filter((market) => !Object.keys(pastNotifiedProposals).includes(getMarketKeyToStore(market)));
 
   // Log the markets that have been proposed but do not have a proposal event.
-  // This is a rare case that should never happen but we want to log it if it does.
   for (const market of proposedMarketsWithoutProposal) {
     const unknownProposalData = getUnknownProposalKeyData(market.question);
     const marketKey = getMarketKeyToStore(unknownProposalData);
 
-    // If we have already logged this market then skip it.
-    if (Object.keys(pastNotifiedProposals).includes(marketKey)) continue;
+    // The first time we see a market proposal event missig we store it as not notified.
+    if (!Object.keys(pastNotifiedProposals).includes(marketKey)) {
+      await storeNotifiedProposals([{ ...unknownProposalData, notified: false }]);
+      continue;
+    }
+
+    const pastNotifiedProposal = pastNotifiedProposals[marketKey];
+    const now = Date.now() / 1000;
+    const timeSinceLastNotification = now - pastNotifiedProposal.notificationTimestamp;
+
+    // By default we only log unknown proposals if we don't find the proposal event after some time.
```

Comment:
So this adds a delay so we only notify on them if they continue to be unknown for a certain period (maybe 30 minutes)?

This is interesting. It's a little different from what we discussed IRL, but I think it's effective as long as these are almost always temporary (is that the case?).

One point, though: in PD today, we _ignore_ these alerts if we get less than some number per hour. If we move to only alerting once on unknown proposals, that might mean that nobody will be alerted on a market where the proposal is permanently unknown, right? So we should remove that PD rule, right?

Reply:
Correct @mrice32, that PD rule needs to be removed before adding this changes to the notifier otherwise we won't get alerted.

Reply:
Sounds good.